sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcDeserializer.scala

```diff
@@ -127,10 +127,12 @@ class OrcDeserializer(
         updater.setInt(ordinal, OrcShimUtils.getGregorianDays(value))
 
       case TimestampType => (ordinal, value) =>
-        updater.setLong(ordinal, DateTimeUtils.fromJavaTimestamp(value.asInstanceOf[OrcTimestamp]))
-
-      case TimestampNTZType => (ordinal, value) =>
-        updater.setLong(ordinal, OrcUtils.fromOrcNTZ(value.asInstanceOf[OrcTimestamp]))
+        if (value.isInstanceOf[OrcTimestamp]) {
+          updater.setLong(ordinal,
+            DateTimeUtils.fromJavaTimestamp(value.asInstanceOf[OrcTimestamp]))
+        } else {
+          updater.setLong(ordinal, value.asInstanceOf[LongWritable].get)
```

Comment:
when can we reach this branch?

Reply:
when we read NTZ as LTZ?

Reply:
then shall we shift the long value w.r.t. the timezone?

Reply:
> when we read NTZ as LTZ?

Yes.